rest/doc_api.go

```diff
@@ -279,7 +279,12 @@ func (h *handler) handlePutDoc() error {
 		if err != nil {
 			return err
 		}
-		newRev = body["_rev"].(string)
+
+		newRevRaw, ok := body["_rev"]
+		if !ok {
+			return base.HTTPErrorf(http.StatusInternalServerError, "Expected revision id in body _rev field")
+		}
+		newRev = newRevRaw.(string)
```

Comment:
I think you could simplify this as 
```
newRev, ok := body["_rev"].(string)
if !ok {
   ...
}
```
https://play.golang.org/p/IMXQ_LdABB

Reply:
This might be tricky.  I started out with something like that (not sure if it was exactly that), but it was creating a new variable rather than updating the existing `newRev` in scope, and leaving the outer `newRev` untouched.

To guard against that possibility, I reworked it to:

```
	        var newRev string
	        var ok bool
		newRev, ok = body["_rev"].(string)
		if !ok {
			return base.HTTPErrorf(http.StatusInternalServerError, "Expected revision id in body _rev field")
		}
```

I guess this is probably not idiomatic, but I've yet to come across a good way to remember the esoteric rules around shadowing and assignment in cases like these